### PR TITLE
Chore: improve error messages to clients

### PR DESCRIPTION
**Pull request by rdong333**

### CODE REVIEW

**Review by melanahammel — APPROVED** [review]

**Comment by mayitbeegh** [comment]
There are a few custom messages that I don't think would be useful to the users. And then returning unvetted user input is potentially vulnerable to xss.

File: cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java [hunk 1]
```diff
@@ -316,11 +319,12 @@ private EncryptedAuthDataWrapper authenticate(
     try {
       authResponseJson = objectMapper.writeValueAsBytes(authResponse);
     } catch (JsonProcessingException e) {
+      String msg = "Failed to write IAM role authentication response as JSON for encrypting.";
```
Comment (mayitbeegh) [inline cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java:322]: The users might get confused by this message.

File: cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java [hunk 2]
```diff
@@ -423,11 +427,12 @@ private AuthTokenResponse createToken(
     try {
       return objectMapper.writeValueAsBytes(authToken);
     } catch (JsonProcessingException e) {
+      String msg = "Failed to write IAM role authentication response as JSON for encrypting.";
```
Comment (mayitbeegh) [inline cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java:429]: The users might get confused by this message.

File: cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java [hunk 3]
```diff
@@ -442,9 +447,11 @@ private AuthTokenResponse createToken(
   public AuthResponse refreshUserToken(final CerberusPrincipal authPrincipal) {
 
     if (!PrincipalType.USER.equals(authPrincipal.getPrincipalType())) {
+      String msg = "The principal: %s attempted to use the user token refresh method";
```
Comment (mayitbeegh) [inline cerberus-web/src/main/java/com/nike/cerberus/service/AuthenticationService.java:450]: The users might get confused by this message.

**Pull request merged**